ext/net/lib.rs

```diff
@@ -9,20 +9,110 @@ pub mod raw;
 pub mod resolve_addr;
 mod tcp;
 
+use deno_core::anyhow::Context;
 use deno_core::error::AnyError;
 use deno_core::OpState;
 use deno_tls::rustls::RootCertStore;
 use deno_tls::RootCertStoreProvider;
+use fqdn::FQDN;
+use std::net::{Ipv4Addr, Ipv6Addr};
 use std::path::Path;
 use std::path::PathBuf;
+use std::str::FromStr;
 use std::sync::Arc;
 
 pub const UNSTABLE_FEATURE_NAME: &str = "net";
 
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
+pub struct NetPermissionHost {
+  pub host: String,
```

Comment:
It's not ideal that we're going from a string to a known struct, to a string and then on each permission check doing the same thing. This code below here is also duplicated in permissions/lib.rs. Is there any way that once we parse something as a FQDN that we leave it as one? It seems like we might need to use an enum here to store multiple different kinds of values (ex. `FQDN` or `IpAddr`)

Reply:
Hi @dsherret 
https://github.com/denoland/deno/pull/23590/commits/dd15f60a8fca9f0652f92867566ecf0bb16ba128
done